src/integration/java/io/pinecone/helpers/TestIndexResourcesManager.java

```diff
@@ -20,7 +20,7 @@ public class TestIndexResourcesManager {
     private static TestIndexResourcesManager instance;
     private static final String apiKey = System.getenv("PINECONE_API_KEY");
     private final int dimension = System.getenv("DIMENSION") == null
-            ? 4
+            ? 3
```

Comment:
Dimension has to == the # of vectors we upsert; so `vectorIds`, i.e. `3`

Reply:
It shouldn't need to match the number of vectors we upsert, just the dimensionality of the vectors themselves, right?

This should be alright to change given the tests grab the dimension from here. If we use 3 though there's a possibility the indexes created by the `TestIndexResourcesManager` could be confused with indexes created by the current `IndexManager` since I think a lot of those indexes have a dimension of 3 and that's how the lookup is done. That'll be going away in my PR (https://github.com/pinecone-io/pinecone-java-client/pull/99), but we may want to leave this set to 4 for now if we can.

Reply:
Oh you're 100% right! I'll change it back to `4` :)